src/control/lib/daos/api/pool.go

```diff
@@ -567,6 +575,80 @@ func PoolDeleteAttributes(ctx context.Context, sysName, poolID string, attrNames
 	return delDaosAttributes(poolConn.daosHandle, poolAttr, attrNames)
 }
 
+func (ph *PoolHandle) ListContainers(ctx context.Context, query bool) ([]*daos.ContainerInfo, error) {
+	if !ph.IsValid() {
+		return nil, ErrInvalidPoolHandle
+	}
+
+	return PoolListContainers(ph.toCtx(ctx), "", "", query)
+}
+
+// PoolListContainers returns a list of information about containers in the pool.
+func PoolListContainers(ctx context.Context, sysName, poolID string, query bool) ([]*daos.ContainerInfo, error) {
+	poolConn, disconnect, err := getPoolConn(ctx, sysName, poolID, daos.PoolConnectFlagReadOnly)
+	if err != nil {
+		return nil, err
+	}
+	defer disconnect()
+	logging.FromContext(ctx).Debugf("PoolListContainers(%s:%t)", poolConn, query)
+
+	if err := ctx.Err(); err != nil {
+		return nil, ctxErr(err)
+	}
+
+	extra_cont_margin := C.size_t(16)
```

Comment:
what happens when this is not big enough and in the event that more than 16 have been created since the first API call? do we get a segfault?

Reply:
or buffer overflow/write to undefined memory?